tsdb/record/record.go

```diff
@@ -107,6 +118,48 @@ func (d *Decoder) Series(rec []byte, series []RefSeries) ([]RefSeries, error) {
 	return series, nil
 }
 
+func (d *Decoder) Exemplars(rec []byte, exemplars []RefExemplar) ([]RefExemplar, error) {
```

Comment:
I would be sorely tempted to not persist exemplars to the WAL as part of this PR - purely to make it smaller, easier to review and therefore merge.